src/device/usbd.c

```diff
@@ -56,6 +56,10 @@ TU_ATTR_WEAK void tud_event_hook_cb(uint8_t rhport, uint32_t eventid, bool in_is
   (void)in_isr;
 }
 
+TU_ATTR_WEAK void tud_sof_cb(uint32_t frame_count) {
+  (void)frame_count;
+}
```

Comment:
Why not have a weak declaration in the header file? After all it's used only once at L629 anyway. So a check there for a definition similar how it's done for other things should be more inline.

Reply:
We are converting to weak default implementation for keil and clang compatibility.